### PR TITLE
#1413 share view enhancement - REDONE

### PR DESCRIPTION
**Pull request by AndyScherzinger**
For details and discussion please see #1413 and original PR #1416

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Screenshots...
device-2016-03-21-203546
device-2016-03-21-203559
device-2016-03-21-203617

**Comment by AndyScherzinger** [comment]
@jancborchardt one thing, the user list with 1.9.2 will have three kinds of item types, users, groups and federated shares. For federated I use the "user" icon at the moment, so we are in need for an icon :smile:

**Comment by jancborchardt** [comment]
@AndyScherzinger it’s fine to use the same icon for federated shares, since we want to make the difference as little as possible to have a similarly seamless UX. In the list display, the username should be displayed like `tobias@…` (with the server really being dotted out, like in the web app).

**Comment by AndyScherzinger** [comment]
Do you have a screenshot for the @... for user and group to see the difference?

**Comment by AndyScherzinger** [comment]
@davivel is there a test account somewhere that I could use to the the federated shares in action and do the possible UI changes?

**Comment by davivel** [comment]
@AndyScherzinger , thanks a lot for redoing this one.

 >  is there a test account somewhere that I could use to the the federated shares 

That's going to be difficult for the short term. And we are interested in this for a very, very, very short term.

> In the list display, the username should be displayed like tobias@… (with the server really being dotted out, like in the web app).

@jancborchardt , we made the list following the web UI, but I can't remember any ```...```. What OC server includes that?

**Comment by AndyScherzinger** [comment]
@davivel no worries, I just wanted to see the user/group/federated display in the web app to better understand Jan's requirement here :)

File: res/layout/edit_share_layout.xml [hunk 1]
```diff
@@ -27,18 +27,40 @@
     <LinearLayout android:orientation="vertical"
                   android:layout_width="match_parent"
                   android:layout_height="wrap_content">
+        <TextView
+            android:id="@+id/editShareTitle"
+            android:layout_width="fill_parent"
```
Comment (davivel) [inline res/layout/edit_share_layout.xml]: Should be ```match_parent```, ```fill_parent``` is old wording, long time deprecated (though harmless)

**Comment by davivel** [comment]
I can upload some screenshots :)

File: res/layout/edit_share_layout.xml (hunk 1 above)
Comment (AndyScherzinger) [inline res/layout/edit_share_layout.xml]: fixed along with some alignment optimizations https://github.com/owncloud/android/pull/1551/commits/968373a647637fe2e82545a86253937cf7cacedd

**Comment by davivel** [comment]
sharing

shared

Captures of web UI while sharing and after shared. I guess the ```...``` talked about are those at the end of user when there is not space enough in a single line (see second picture).

**Comment by davivel** [comment]
Code is :+1:

@AndyScherzinger , this is your chance to make a very easy rebase ;-) 

```git pull
   git checkout 1413_share_view_enhancement_redo
   git rebase master ```

**Comment by davivel** [comment]
cc @jesmrec , ready to test.

**Comment by AndyScherzinger** [comment]
:sad:
```git rebase master``` ends up with: 

```fatal: Needed a single revision```
```invalid upstream master```

**Comment by davivel** [comment]
Do you have local copy of master?

Try ```git checkout master```, then back to ```git checkout 1413_share_view_enhancement_redo```

**Comment by AndyScherzinger** [comment]
Hmmmm, I don't get it... Tried what you said, see commit history, again all commits doubled...

**Comment by AndyScherzinger** [comment]
@jancborchardt looking at the screenshots this might be a really tricky one since we don't have that much space on a cellphone, right?
device-2016-03-21-203546
device-2016-03-21-203559

**Comment by AndyScherzinger** [comment]
Else we might be able to do two lines and add that to a second line - ?

**Comment by davivel** [comment]
Two lines not, please... 

Let me see how it's now, I'll push a screenshot with a federated share.

**Comment by AndyScherzinger** [comment]
I'm fine with it the way it is, question is, if we pack in the server address + port it'll be very packed. That is just my concern here.

**Comment by davivel** [comment]
> Hmmmm, I don't get it... Tried what you said, see commit history, again all commits doubled...

Don't worry, I rebased. This was not so tricky, since there was no conflict. 


@jancborchardt , screenshot of federated share:
device-2016-03-22-133737

Ellipsis is done in the middle, a common practice in our mobile apps, if I recall correctly. Is it OK, or should we ellipsize in the end ?

**Comment by AndyScherzinger** [comment]
Looks good to me :+1: 
Personally I would stick with the ellipsis in the middle like @davivel said this common practice in the mobile app and ellipsizing at the end but still having the port definition makes things a bit complicated just for the display of the more or less irrelevant information from an end user's perspective (port number).

**Comment by AndyScherzinger** [comment]
@carlaschroder just in case this is relevant for documentation, please see above the latest screenshots of the share dialog for the upcoming release 1.9.2 :smiley:

**Comment by javiergonzper** [comment]
QA approved :+1:

**Comment by davivel** [comment]
Then... in!

**Comment by AndyScherzinger** [comment]
AWESOME! :tada:

**Comment by jancborchardt** [comment]
Nice, very cool! :)

**Pull request merged**